opentelemetry-exporter-gcp-trace/src/opentelemetry/exporter/cloud_trace/__init__.py

```diff
@@ -88,16 +89,22 @@ class CloudTraceSpanExporter(SpanExporter):
         project_id: ID of the cloud project that will receive the traces.
         client: Cloud Trace client. If not given, will be taken from gcloud
             default credentials
+        use_otel_env_resource: A boolean to extract resource attributes
+            that were added via special OpenTelemetry variables such as
+            `OTEL_RESOURCE_ATTRIBUTES` (default: False).
     """
 
     def __init__(
-        self, project_id=None, client=None,
+        self, project_id=None, client=None, use_otel_env_resource=False,
     ):
         self.client = client or TraceServiceClient()
         if not project_id:
             _, self.project_id = google.auth.default()
         else:
             self.project_id = project_id
+        self.otel_env_resource = (
+            OTELResourceDetector().detect() if use_otel_env_resource else None
+        )
```

Comment:
This isn't needed, this is done automatically by OTel. The resource this exporter sees below will already have the fields from the environrment variable.

I was thinking we could just add a regex here, where resource keys that match the regex will be passed on as span attributes.

Reply:
But we don't know what the resource keys are because they are based on the environmental variable.  So you want only a few pre-set list of resource attribute to be passed on to the span attribute?  Doesn't that kind of defeat the purpose of allowing people to use the `OTEL_RESOURCE_ATTRIBUTES` environmental variable?

Reply:
I could just extract the keys here, but I thought it made a little more sense for all the inputs to `_extract_resources` to be resources instead of a hybrid between `Resource` and `Attribute` / `Dict[str, str]`.  The other idea is to add it to `resources_and_attrs` in line 166 directly with `**self.otel_env_resource`, but then testing would require a bit of refactoring.

Reply:
In my use case, I gave some examples to simplify what I actually wanted.  The truth is we use a non-standard name called `app_name` and I would probably use that instead of `service.name`.  But I didn't want this PR to be able my specific case, but more about generally adding extra attributes to the spans.  The idea being that one could add arbitrary attributes (even ridiculously named non-standard attributes like `app_name`) that would show up in the GCP interface as labels which can be used for filtering by using the environmental variable `OTEL_RESOURCE_ATTRIBUTES`.

Apologies if I wasn't clear on this earlier.

Reply:
No worries. For the general use case, I think we can address it again later if someone can't solve it with a more restrictive regex.

Would either a filter function that returns true/false if the resource key should be included, or a regex pattern for resource keys to include solve your specific issue?

Reply:
Ok, I get it.  Yea, that would work for me.  I'll implement the regex.